pkg/operator/ceph/cluster/osd/topology/topology.go

```diff
@@ -116,6 +116,20 @@ func extractTopologyFromLabels(labels map[string]string) (map[string]string, str
 			topologyAffinity = formatTopologyAffinity(label, value)
 		}
 	}
+
+	// iterate in lowest to highest order as the lowest level should be sustained and higher level duplicate
+	// should be removed
+	duplicateTopology := make(map[string]int)
```

Comment:
Instead of creating a new loop, how about keeping track of this in the loop on lines 111-118 and then just not add it to `topology[topology]` on line 115 if it already exists?

Reply:
Actually I would have done that only, 
But that loop is running in reverse order,

And according to the current implementations or region and zone I feel like we need to sustain the lower hierarchy duplicates and delete the higher level duplicates,
So for that, we need an extra loop.

If anything irrespective of hierarchy can be removed I can change it like that